modules/basics/src/main/java/com/opengamma/strata/basics/currency/Money.java

```diff
@@ -0,0 +1,257 @@
+   * <p>
+   * The result will be expressed in terms of the given currency, converting
+   * using the specified FX rate.
+   * <p>
+   * For example, if this represents 'GBP 100' and this method is called with
+   * arguments {@code (USD, 1.6)} then the result will be 'USD 160'.
+   *
+   * @param resultCurrency the currency of the result
+   * @param fxRate the FX rate from this currency to the result currency
+   * @return the converted instance, which should be expressed in the specified currency
+   * @throws IllegalArgumentException if the FX is not 1 when no conversion is required
+   */
+  public Money convertedTo(Currency resultCurrency, BigDecimal fxRate) {
+    if (currency.equals(resultCurrency)) {
+      if (DoubleMath.fuzzyEquals(fxRate.doubleValue(), 1d, 1e-8)) {
+        return this;
+      }
+      throw new IllegalArgumentException("FX rate must be 1 when no conversion required");
+    }
+    return Money.of(resultCurrency, amount.multiply(fxRate));
+  }
+
+  /**
+   * Converts this amount to an equivalent amount in the specified currency.
+   * <p>
+   * The result will be expressed in terms of the given currency.
+   * If conversion is needed, the provider will be used to supply the FX rate.
+   *
+   * @param resultCurrency the currency of the result
+   * @param rateProvider the provider of FX rates
+   * @return the converted instance, in the specified currency
+   * @throws RuntimeException if no FX rate could be found
+   */
+  @Override
+  public Money convertedTo(Currency resultCurrency, FxRateProvider rateProvider) {
+    if (currency.equals(resultCurrency)) {
+      return this;
+    }
+    double converted = rateProvider.convert(amount.doubleValue(), currency, resultCurrency);
+    return Money.of(resultCurrency, converted);
+  }
+
+  /**
+   * Compares this instance of {@link Money} to another.
+   * <p>
+   * This compares currencies alphabetically, then by amount.
+   *
+   * @param other the other amount
+   * @return negative if less, zero if equal, positive if greater
+   */
+  @Override
+  public int compareTo(Money other) {
+    return ComparisonChain.start()
+        .compare(currency, other.currency)
+        .compare(amount, other.amount)
+        .result();
+  }
+
+  /**
+   * Parses the string to produce a {@link Money}.
+   * <p>
+   * This parses the {@code toString} format of '${currency} ${amount}'.
+   *
+   * @param amountStr  the amount string
+   * @return the currency amount
+   * @throws IllegalArgumentException if the amount cannot be parsed
+   */
+  @FromString
+  public static Money parse(String amountStr) {
+    ArgChecker.notNull(amountStr, "amountStr");
+    List<String> split = Splitter.on(' ').splitToList(amountStr);
+    if (split.size() != 2) {
+      throw new IllegalArgumentException("Unable to parse amount, invalid format: " + amountStr);
+    }
+    try {
+      Currency cur = Currency.parse(split.get(0));
+      double amount = Double.parseDouble(split.get(1));
```

Comment:
Probably better to create the `BigDecimal` directly from the string rather than creating a `double` (which might not be able to exactly express the amount) and then creating a `BigDecimal` from that. Use `new BigDecimal(split.get(1));`